### PR TITLE
[WIP]Expose replace methods for related models

### PR DESCRIPTION
**Pull request by ivanschwarz**
### Description

This PR is a prerequisite for strongloop/loopback#3959. You can find details there.

@bajtos : Can you review?

#### Related issues

- Reported in strongloop/loopback#3139

### Checklist

- [ ] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style
  guide](http://loopback.io/doc/en/contrib/style-guide.html)

### CODE REVIEW

**Comment by slnode** [comment]
Can one of the admins verify this patch? To accept patch and trigger a build add comment ".*ok\W+to\W+test.*"

**Comment by ivanschwarz** [comment]
@bajtos : Thank you for your review. I think we should wait with merging this PR. Like I noted in strongloop/loopback#3959, I've implemented changes just for `hasMany` relation. I think I should extend this functionality to all other relevant relations. Essentially every PUT endpoint should eventually have its matching PATCH endpoint. I just wanted to have a positive feedback on my approach first to avoid unnecessary work.

File: test/relations.test.js [hunk 1]
```diff
@@ -445,6 +469,34 @@ describe('relations', function() {
...
               ch.name.should.equal('aa');
+              ch.type.should.equal('b');
+              done();
+            });
+        }
+      });
+
+      it('should replace scoped record with promises', function(done) {
+        var id;
+        Book.create()
```
Comment (jannyHou) [inline test/relations.test.js]: nitpick(Not a big deal feel free to ignore): To simplify the test code I think you can omit `done` here.
```js
it('should replace...', function() {
  return Book.create()
    .then(function(book) {
      // create chapters
    })
    .then(function(ch) {
       return verify(book);
     });

    function verify(book) {
      return book.chapters.findById(id)
        .then(function(ch) {
          // verify chapter
         });
    }
})
```

**Comment by raymondfeng** [comment]
@jannyHou PTAL

**Pull request closed without merging**